Assets/MixedRealityToolkit/Interfaces/Services/IMixedRealityDataProviderAccess.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+    public interface IMixedRealityDataProviderAccess
+    {
+        /// <summary>
+        /// Gets the collection of registered data providers.
+        /// </summary>
+        /// <returns>
+        /// Read only copy of the list of registered observers.
+        /// </returns>
+        IReadOnlyList<IMixedRealityDataProvider> GetDataProviders();
+
+        /// <summary>
+        /// Get the collection of registered observers of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The desired data provider type</typeparam>
+        /// <returns>
+        /// Readonly copy of the list of registered data providers that implement the specified type.
+        /// </returns>
+        IReadOnlyList<T> GetDataProviders<T>() where T : IMixedRealityDataProvider;
+
+        /// <summary>
+        /// Get the data provder that is registered under the specified name.
+        /// </summary>
+        /// <param name="name">The friendly name of the data provider.</param>
+        /// <returns>
+        /// The requested data provider, or null if one cannot be found.
+        /// </returns>
+        /// <remarks>
+        /// If more than one data provider is registered under the specified name, the first will be returned.
+        /// </remarks>
+        IMixedRealityDataProvider GetDataProvider(string name);
+
+        /// <summary>
+        /// Get the data provider that is registered under the specified name matching the specified atch type.
+        /// </summary>
+        /// <typeparam name="T">The desired data provider type.</typeparam>
+        /// <param name="name">The friendly name of the data provider.</param>w
+        /// <returns>
+        /// The requested data provider, or null if one cannot be found.
+        /// </returns>
+        /// <remarks>
+        /// If more than one data provider is registered under the specified name, the first will be returned.
+        /// </remarks>
+        T GetDataProvider<T>(string name = null) where T : IMixedRealityDataProvider;
```

Comment:
I don't fully understand why one of these supports taking a null name, but the other doesn't - what's the rationale behind this non-consistent API story?

Reply:
Are you referring to `IMixedRealityDataProvider GetDataProvider(string name);` vs `T GetDataProvider<T>(string name = null)`?

The reason, as I understand it, is that both calls need something to actually let you define what you're looking for. The first call lets you ask by name, while the second one lets you ask by type and an optional name.